### DISCUSSION TITLE
limit site access to conf host only, no IP access

### DISCUSSION THREAD

**Discussion by badiku** [RealTimeLogic/BAS#12] [Q&A] [answered]
hi,

 if I set host='mysite.net' in mako.conf,
is it possible to limit site access to this host only, no IP access?

only http://mysite.net
not  http://198.18.28.x

or use sslhost to limit https

maybe instead of using mako -l to load a site, I need to setup a site by some ba function manually?

...

ok,  later I found that maybe I should use `domainresrdr`, 

https://realtimelogic.com/ba/doc/?url=lua.html#ba_create_domainresrdr

my .preload:
```` 
print('configure virtual hosts')
dir:unlink()
local domains = {
   "my-domain-a.com",
   "my-domain-b.com",
   "my-domain-c.com"
}
local hio = ba.openio"home"
dirs = {}
for _, domain in pairs(domains) do
   print('domain:', domain)
   assert(hio:stat(domain) or hio:mkdir(domain))
   local dir = ba.create.domainresrdr(domain, ba.mkio(hio, domain))
   local appenv=setmetatable({io=io,dir=dir}, {__index=_G})
   appenv.app = appenv
   dir:lspfilter(appenv)
   dir:insert()
   dirs[domain] = dir
end
app.dirs = dirs
```

since ip is not in specified domains, it always returns "404 not found"

any better idea?





and BTW, by checking .config, I find that there's some code about mako.setzippwd

maybe it is a hidden undocumented feature about using password protected mako.zip?

**Comment by surfskidude** [comment]
If you want to host multiple domains, use domainresrdr. For reference, I have attached a .preload script we use for our online server, powered by FuguHub.
[preload.txt](https://github.com/RealTimeLogic/BAS/files/13514502/preload.txt)

**Reply by badiku** [reply]
nice. thank you for sharing.

is fuguhub a fixed old version unchanged several years ago, 
and makoserver is a new version keeping updated?

BTW, 
certificate "Real Time Logic Root CA" has expired on 2023.10.2

https://realtimelogic.com/downloads/root-certificate/

https://realtimelogic.com/downloads/certificates/CA_RTL_RSA_2048.cer

>All SharkSSL enabled products and demos delivered by Real Time Logic include a certificate that is digitally signed by Real Time Logic's root certificate. This certificate is not in your browser's certificate store unless you have previously installed it. The following instructions are for installing Real Time Logic's root certificate.

**Reply by surfskidude** [reply, marked as answer]
Mako is designed for developers, FuguHub can be used by anyone.
